cmd/prober/prober.go

```diff
@@ -104,44 +108,118 @@ func encodeLevel() zapcore.LevelEncoder {
 }
 
 var (
-	logStyle       string
-	frequency      int
+	scPath      string
+	trPath      string
+	trustedRoot *root.TrustedRoot
+	staging     bool
+
+	frequency int
+	logStyle  string
+	addr      string
+	grpcPort  int
+
 	retries        uint
-	addr           string
-	rekorURL       string
-	fulcioURL      string
-	fulcioGrpcURL  string
-	tsaURL         string
 	oneTime        bool
 	runWriteProber bool
-	versionInfo    version.Info
+
+	rekorV1Services []root.Service
+	fulcioService   root.Service
+	fulcioGrpcURL   string
+	tsaServices     []root.Service
+
+	rekorRequestsJSON  string
+	fulcioRequestsJSON string
+
+	versionInfo version.Info
+
+	err error
 )
 
 type attemptCtxKey string
 
 func init() {
+	flag.StringVar(&scPath, "signing-config", "", "Path to the signing config")
+	flag.StringVar(&trPath, "trusted-root", "", "Path to the trusted root")
+
+	flag.BoolVar(&staging, "staging", false, "Whether to use the public instance staging environment (otherwise use the public instance production environment). For private deployments, use the signing-config and trusted-root flags.")
+
 	flag.IntVar(&frequency, "frequency", 10, "How often to run probers (in seconds)")
-	flag.StringVar(&logStyle, "logStyle", "prod", "log style to use (dev or prod)")
+	flag.StringVar(&logStyle, "logStyle", "prod", "Log style to use (dev or prod)")
 	flag.StringVar(&addr, "addr", ":8080", "Port to expose prometheus to")
+	flag.IntVar(&grpcPort, "grpc-port", 5554, "Port for Fulcio gRPC endpoint (only if configured)")
+
+	flag.UintVar(&retries, "retry", 4, "Maximum number of retries before marking HTTP request as failed")
+	flag.BoolVar(&oneTime, "one-time", false, "Whether to run only one time and exit")
+	flag.BoolVar(&runWriteProber, "write-prober", false, " [Kubernetes only] run the probers for the write endpoints")
```

Comment:
Is this still accurate to only run on K8s? It now works locally and against the hermetic environment, correct?